src/Http/Http.Extensions/gen/Microsoft.AspNetCore.Http.ValidationsGenerator/Parsers/ValidationsGenerator.TypesParser.cs

```diff
@@ -89,15 +89,74 @@ internal bool TryExtractValidatableType(ITypeSymbol typeSymbol, RequiredSymbols
     internal ImmutableArray<ValidatableProperty> ExtractValidatableMembers(ITypeSymbol typeSymbol, RequiredSymbols requiredSymbols, ref HashSet<ValidatableType> validatableTypes, ref List<ITypeSymbol> visitedTypes)
     {
         var members = new List<ValidatableProperty>();
+        var resolvedRecordProperty = new List<IPropertySymbol>();
+
+        // Special handling for record types to extract properties from
+        // the primary constructor.
+        if (typeSymbol is INamedTypeSymbol { IsRecord: true } namedType)
+        {
+            // Find the primary constructor for the record, account
+            // for members that are in base types to account for
+            // record inheritance scenarios
+            var primaryConstructor = namedType.Constructors
+                .FirstOrDefault(c => c.Parameters.Length > 0 && c.Parameters.All(p =>
+                    namedType.FindPropertyIncludingBaseTypes(p.Name) != null));
+
+            if (primaryConstructor != null)
+            {
+                // Process all parameters in constructor order to maintain parameter ordering
+                foreach (var parameter in primaryConstructor.Parameters)
+                {
+                    // Find the corresponding property in this type, we ignore
+                    // base types here since that will be handled by the inheritance
+                    // checks in the default ValidatableTypeInfo implementation.
+                    var correspondingProperty = typeSymbol.GetMembers()
+                        .OfType<IPropertySymbol>()
+                        .FirstOrDefault(p => string.Equals(p.Name, parameter.Name, System.StringComparison.OrdinalIgnoreCase));
+
+                    if (correspondingProperty != null)
+                    {
+                        resolvedRecordProperty.Add(correspondingProperty);
+
+                        // Check if the property's type is validatable, this resolves
+                        // validatable types in the inheritance hierarchy
+                        var hasValidatableType = TryExtractValidatableType(
+                            correspondingProperty.Type.UnwrapType(requiredSymbols.IEnumerable),
+                            requiredSymbols,
+                            ref validatableTypes,
+                            ref visitedTypes);
+
+                        members.Add(new ValidatableProperty(
+                            ContainingType: correspondingProperty.ContainingType,
+                            Type: correspondingProperty.Type,
+                            Name: correspondingProperty.Name,
+                            DisplayName: parameter.GetDisplayName(requiredSymbols.DisplayAttribute) ??
+                                        correspondingProperty.GetDisplayName(requiredSymbols.DisplayAttribute),
+                            Attributes: []));
```

Comment:
Should we be populating the properties attributes here?

Reply:
This property was useful when we were statically resolving the attributes and source-generating the cache. Since we don't do that anymore, I opted not to do it here.

Technically, we can also remove it from the other initializers for this type as well but I'll put a note as to why this one is empty. It's fairly easy to add these back in if we decide to resolve more attributes statically.